chain/index/api.go

```diff
@@ -0,0 +1,167 @@
+package index
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+func (si *SqliteIndexer) sanityCheckBackfillEpoch(ctx context.Context, epoch abi.ChainEpoch) error {
+	// should be less than the max non reverted height in the Index
+	var maxNonRevertedHeight sql.NullInt64
+	err := si.stmts.getMaxNonRevertedHeightStmt.QueryRowContext(ctx).Scan(&maxNonRevertedHeight)
+	if err != nil {
+		return xerrors.Errorf("failed to get max non reverted height: %w", err)
+	}
+	if !maxNonRevertedHeight.Valid {
```

Comment:
```suggestion
	// couldn't find any non-reverted entries
	if !maxNonRevertedHeight.Valid {
```

that's what this is doing, right?